generic/proof-shell.el

```diff
@@ -1193,7 +1193,8 @@ contains only invisible elements for Prooftree synchronization."
 	     (or (null proof-action-list)
 		 (cl-every
 		  (lambda (item) (memq 'proof-tree-show-subgoal (nth 3 item)))
-		  proof-action-list)))))))
+		  proof-action-list)
+		 (string= (car (nth 1 item)) "Show.")))))))
```

Comment:
Good point. At first sight you'll just have to replace this with something like:

```suggestion
		 (string-match-p proof-show-proof-diffs-regexp (last (nth 1 item)))))))))
```

(namely, we don't test that the "Show" (which is not inserted every time) is at the beginning (car), but that "Show Proof" is at the end (last)